src/actions/heap/heap.ts

```diff
@@ -165,29 +181,46 @@ export class HeapAction extends Hub.Action {
         errors.length === 0 ? "success" : "failure",
       )
     } catch (err) {
-      winston.warn("Heap track call failed.")
+      winston.warn("Heap track call failed.", {
+        ...logTag,
+        error: getErrorMessage(err),
+      })
       // swallow internal track call error
     }
 
     if (errors.length === 0) {
       return new Hub.ActionResponse({ success: true })
+    } else {
+      // limit error message to the first N to avoid returning enormous error messages
+      // (arbitrary limit)
+      const errorsToDisplay = errors.slice(0, HeapAction.DISPLAY_ERROR_COUNT)
+      // tell how many errors there were in total since we're only displaying the first N
+      const errorDesc = `Heap action for envId ${envId} failed with ${errors.length} errors ` +
+        (errors.length > HeapAction.DISPLAY_ERROR_COUNT ? `(displaying first ${HeapAction.DISPLAY_ERROR_COUNT})` : "")
+      winston.error(errorDesc, logTag)
+      // log first N errors
+      errorsToDisplay.forEach((err) => winston.error(`envId ${envId} error: ${err.message}`, logTag))
+      // concat first N errors into a signle errorMsg to return to the looker action hub.
+      const errorMsg = errorsToDisplay.map((err) => err.message).join(", ")
```

Comment:
filter out InternalError in the next PR. I want to deploy this version and see the response firstly, see if it's improved.